Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/service/PersistedTasksService.java

```diff
@@ -203,23 +186,61 @@ public Action0 clearCompletedTasks() {
         return new Action0() {
             @Override
             public void call() {
+                Event<Tasks> tasksEvent = taskRelay.getValue();
+                Tasks value = tasksEvent.data().or(Tasks.empty());
+                Tasks uncompletedTasks = Tasks.from(ImmutableList.copyOf(Iterables.transform(value.all(), markCompletedAsDeleted())));
```

Comment:
`Tasks.from(ImmutableList.copyOf(...` is used twice in this PR only, may be worth extracting a new method that copies the input list also to reduce nested calls.